lib/puma/runner.rb

```diff
@@ -194,5 +194,12 @@ def stats
         }
       }
     end
+
+    # this method call should always be guarded by `@log_writer.debug?`
+    def debug_loaded_extensions(str)
```

Comment:
Why not have the guard in this method?

Also, I think "Puma running in debug mode" (`PUMA_DEBUG` set) is not something only `LogWriter` should know about. In the future, more things could be happening in "debug more" (e.g. collecting more details stats) in my mind.

What do you think of adding a constant `PUMA_DEBUG` to `lib/puma.rb`? And a method `Puma.debug?` (or `Puma.debug_mode?`) and use that here. That method can also be used in both `LogWriter` and `ErrorLogger` (both do `ENV.key?('PUMA_DEBUG')` today)

Reply:
I'm fine with either option, up to Greg.